src/shared/parse-helpers.c

```diff
@@ -196,3 +200,86 @@ int parse_socket_bind_item(
         *port_min = mn;
         return 0;
 }
+
+int open_file_parse(const char *v, OpenFile **ret) {
+        _cleanup_free_ char *options = NULL;
+        _cleanup_(open_file_freep) OpenFile *of = NULL;
+        int r;
+
+        assert(v);
+        assert(ret);
+
+        of = new0(OpenFile, 1);
+        if (!of)
+                return -ENOMEM;
+
+        r = extract_many_words(&v, ":", EXTRACT_DONT_COALESCE_SEPARATORS, &of->path, &of->fdname, &options, NULL);
+        if (r < 0)
+                return r;
+
+        if (r == 0)
+                return -EINVAL;
+
+        if (!path_is_absolute(of->path))
+                return -EINVAL;
+
+        if (isempty(of->fdname)) {
+                r = free_and_strdup(&of->fdname, of->path);
+                if (r < 0)
+                        return r;
+        }
+
+        for (const char *p = options;;) {
+                OpenFileFlags flag;
+                _cleanup_free_ char *word = NULL;
+
+                r = extract_first_word(&p, &word, ",", 0);
+                if (r < 0)
+                        return r;
+
+                if (r == 0)
+                        break;
+
+                flag = open_file_flags_from_string(word);
+                if (flag < 0)
+                        return flag;
+
+                of->flags |= flag;
+        }
+
+        *ret = TAKE_PTR(of);
+
+        return 0;
+}
+
+const char *open_file_to_string(const OpenFile *of) {
+        _cleanup_free_ char *options = NULL;
+
+        if (!of)
+                return "";
+
+        for (OpenFileFlags flag = OPENFILE_RDONLY; flag < _OPENFILE_MAX; flag <<= 1) {
+                if ((flag & ~_OPENFILE_MASK_PUBLIC) != 0)
+                        continue;
+
+                if (FLAGS_SET(of->flags, flag) && !strextend_with_separator(&options, ",", open_file_flags_to_string(flag)))
+                                return NULL;
+        }
+
+        return strjoin(of->path, ":", of->fdname, ":", options);
+}
+
+OpenFile* open_file_free(OpenFile *of) {
+        if (!of)
+                return NULL;
+
+        free(of->path);
+        free(of->fdname);
+        return mfree(of);
+}
+
+static const char* const open_file_flags_table[_OPENFILE_MAX] = {
+        [OPENFILE_RDONLY] = "ro",
```

Comment:
what about write-only?

I'd probably support truncate and append here too, just to have something to test

Reply:
hmm, and I think it's important to have a flag that makes it non-fatal if we can't open a file. i.e. "graceful" or so.

Reply:
Thanks for the review!
I don't see any harm in supporting the `append` and `truncate` flags but I will check :slightly_smiling_face: .
For the non-fatal flag, I think it's a good idea. Maybe an `"ig"` flag (for `ignore`)?

Reply:
For `ignore` we can just use the full name, there's no need to use a two letter abbreviation for every option